uefi-raw/src/protocol/misc.rs

```diff
@@ -22,3 +23,30 @@ pub struct TimestampProperties {
     /// example, a 24-bit counter would have an end value of `0xff_ffff`.
     pub end_value: u64,
 }
+
+/// Properties of Reset Notification.
+#[derive(Debug)]
+#[repr(C)]
+pub struct ResetNotificationProtocol {
+    pub register_reset_notify: unsafe extern "efiapi" fn(
+        this: *const Self,
+        reset_function: Option<ResetSystemFn>,
+    ) -> Status,
+    pub unregister_reset_notify: unsafe extern "efiapi" fn(
+        this: *const Self,
+        reset_function: Option<ResetSystemFn>,
+    ) -> Status,
+}
+
+impl ResetNotificationProtocol {
+    pub const GUID: Guid = guid!("9da34ae0-eaf9-4bbf-8ec3-fd60226c44be");
+}
+
+/// Raw reset notification function, to be called if you register it when a RestSystem() is executed.
+// copy from uefi-raw/src/table/runtime.rs:53  at commit@6093205c3eb27b2e78be4c003c04d46679bff420
```

Comment:
Rather than copying it, let's add this type definition in `table/runtime.rs`, and use it in the runtime table too.

Reply:
A little different here,  the  function from uefi-raw/src/table/runtime.rs:53 has a return type "!".
```rust
   pub reset_system: unsafe extern "efiapi" fn(
        rt: ResetType,
        status: Status,
        data_size: usize,
        data: *const u8,
    ) -> !,
```

Reply:
Then why shouldn't the high-level wrapper not also return `!`? IMHO it is odd or even UB if you expect the control flow to continue when it is guaranteed.

Ahh, it might be that the call to the underlying function fails, right?

Would `Result<!, Error>` be an option?